packages/deployment/src/deployers/FaucetDeployer.js

```diff
@@ -0,0 +1,68 @@
+const { fromWei } = require('web3-utils')
+
+const BaseDeployer = require('./BaseDeployer')
+const logger = require('../helpers/logger')('FaucetDeployer')
+
+module.exports = class extends BaseDeployer {
+  constructor(config, environment, output) {
+    super(environment, output)
+    this.config = config
+  }
+
+  async call() {
+    const address = this.previousDeploy.address
+    const ERC20Faucet = await this.environment.getArtifact('ERC20Faucet', '@aragonone/erc20-faucet')
+
+    return address
+        ? this._loadFaucet(ERC20Faucet, address)
+        : this._deployFaucet(ERC20Faucet)
+  }
+
+  async _loadFaucet(ERC20Faucet, address) {
+    logger.warn(`Using previous deployed faucet instance at ${address}`)
+    return ERC20Faucet.at(address)
+  }
+
+  async _deployFaucet(ERC20Faucet) {
+    const sender = await this.environment.getSender()
+    this._printFaucetDeploy(sender)
+    const { owner, tokens } = this.config
+
+    const tokensAddresses = tokens.map(token => token.address)
```

Comment:
Wondering if we should use the `ERC20Deployer` here as well? I don't really foresee a case where you'd want to deploy the faucet first, but it might be useful to either:

- Document that `deploy-faucet` is dependent on a previous protocol deploy, or
- Allow this to happen on its own